psi4/driver/driver_cbs.py

```diff
@@ -703,6 +893,149 @@ def return_energy_components():
+        elif nbasis == 3:
+            return scf_xtpl_helgaker_3
+        else:
+            raise ValidationError("Wrong number of basis sets supplied to scf_xtpl: %d" % nbasis)
+    elif xtpl_type ==  "corl":
+        if nbasis == 2:
+            return corl_xtpl_helgaker_2
+        else:
+            raise ValidationError("Wrong number of basis sets supplied to corl_xtpl: %d" % nbasis)
+            
+
+def _process_cbs_kwargs(kwargs):
+    """ A helper function which either translates supplied kwargs into the
+    ``cbs_metadata`` format, or validates a provided cbs_metadata array.
+    
+    Parameters
+    ----------
+    kwargs : dict
+        kwargs containing the CBS function specification.
+    
+    Returns
+    -------
+    array
+        Array of dictionaries, with each item consisting of an extrapolation
+        stage. All validation takes place here.
+    """    
+    
+    metadata = []
+    if "cbs_metadata" in kwargs:
+        cbs_metadata = kwargs.get("cbs_metadata")
+        for item in cbs_metadata:
+            metadata.append({})
+            if "wfn" in item:
+                if "wfn" not in item:
+                    raise ValidationError("Wavefunction type has to be supplied for every stage.")
+                else:
+                    metadata[-1]["wfn"] = item.get("wfn")
+                if "basis" not in item:
+                    raise ValidationError("Basis set has to be supplied for every stage.")
+                else:
+                    metadata[-1]["basis"] = _expand_bracketed_basis(item.get("basis").lower())
+                metadata[-1]["treatment"] = item.get("treatment", "scf" if len(metadata) == 1 else "corl")
+                metadata[-1]["scheme"] = item.get("scheme", _get_default_xtpl(len(metadata[-1]["basis"][1]),
+                                                                                 metadata[-1]["treatment"])) 
+                if len(metadata) > 1:
+                    metadata[-1]["wfn_lo"] = item.get("wfn_lo", metadata[-2].get("wfn"))
+                    metadata[-1]["basis_lo"] = _expand_bracketed_basis(item.get("basis_lo", 
+                                                                       item.get("basis")).lower())
+                    if len(metadata[-1]["basis"][0]) != len(metadata[-1]["basis_lo"][0]):
+                        raise ValidationError("""Number of basis sets inconsistent 
+                                             between high ({0:d}) and low ({1:d}) levels.""".format(len(metadata[-1]["basis"][0]),len(metadata[-1]["basis_lo"][0])))
+                metadata[-1]["stage"] = item.get("stage", None)
+                if metadata[-1]["stage"] == None:
+                    if len(metadata) == 1:
+                        metadata[-1]["stage"] = "scf"
+                    elif len(metadata) == 2:
+                        metadata[-1]["stage"] = "corl"
+                    else:
+                        metadata[-1]["stage"] = "delta{0:d}".format(len(metadata) - 2)
```

Comment:
If you're doing the `0` in `delta{0:d}` in order to specify the item in the format string to format, that was only necessary in py26 or if you want to go out-of-order. 

You may have a better reason -- I'm not totally up on format string syntax.